rs/registry/nns_data_provider/src/registry.rs

```diff
@@ -312,6 +315,42 @@ impl RegistryCanister {
             ))]),
         }
     }
+
+    pub async fn swap_node_in_subnet_directly(
```

Comment:
This is outside of the scope of the nns_data_provider crate, which is meant to provide tools to read from the registry, not to interact with it. 

This should all be done directly in ic-admin.

You could still use the RegistryCanister, and make ' choose_random_agent' public, and then move this logic into the ic-admin.  That also removes the need to import the types into nns_data_provider, which would affect a lot of other builds in the whole stack, which currently don't need to be rebuilt when registry changes.